src/main/java/org/apache/commons/compress/archivers/tar/TarArchiveInputStream.java

```diff
@@ -243,8 +258,16 @@ public long skip(final long n) throws IOException {
             return 0;
         }
 
-        final long available = entrySize - entryOffset;
-        final long skipped = IOUtils.skip(is, Math.min(n, available));
+        long available;
+        long skipped;
+        if(!currEntry.isSparse()) {
```

Comment:
This is creating a lot of duplicate code paths. Maybe the sparse handling could be just moved into get `getRealSize()` and we can keep the single code path?

Reply:
I agree. I will modify this and move the handling to the `getRealSize`.